lib/c/criu.c

```diff
@@ -47,10 +61,11 @@ void criu_set_service_comm(enum criu_service_comm comm)
 
 void criu_local_set_service_address(criu_opts *opts, const char *path)
 {
+	criu_free_service(opts);
 	if (path)
-		opts->service_address = path;
+		opts->service_address = strdup(path);
```

Comment:
strdup() can return NULL, do we need to handle this?

Reply:
There are more places where we don't check returns codes of strdup. Could you send another patch which will fix all of them?

Reply:
This fix is added in #606. (0ed4db3)
But note, that it requires the change of the return type of the affected functions.